shoulder/generator/cxx_header_generator.py

```diff
@@ -48,350 +46,403 @@ def generate(self, objects, outpath):
-        reg_getter = reg_getter.format(
-            indent = self._indent_string(),
-            size_t = reg_cxx_size,
-            funcname = config.register_read_function,
-            regname = reg_cxx_name
-        )
-        outfile.write(reg_getter)
-
-        reg_setter = "{indent}inline void {funcname}({size_t} val) noexcept "
-        reg_setter += "{{ SET_SYSREG_BY_VALUE_FUNC({regname}, val) }}\n"
-        reg_setter = reg_setter.format(
-            indent = self._indent_string(),
-            funcname = config.register_write_function,
-            size_t = reg_cxx_size,
-            regname = reg_cxx_name
-        )
-        outfile.write(reg_setter)
+        @cxx.namespace(name=str(reg.name).lower())
+        def _in_register_namespace(self, outfile, reg):
+            if reg.size == 32:
+                self._generate_sysreg_32_get(outfile, reg)
+                self._generate_sysreg_32_set(outfile, reg)
 
-        self._generate_fieldsets(reg, outfile)
-
-        # TODO: Make Bareflank independent printing function
-        outfile.write("\n")
-        dump_func = "{indent}inline void dump(int level, std::string *msg = nullptr)\n{indent}{{\n"
-        dump_func = dump_func.format(
-            indent = self._indent_string()
-        )
-        outfile.write(dump_func)
-
-        self._increase_indent()
-        dump_func = "{indent}bfdebug_nhex(level, name, get(), msg);\n"
-        for idx, fieldset in enumerate(reg.fieldsets):
-            for field in fieldset.fields:
-                field_cxx_name = field.name.lower()
-                dump_func += "{indent}{fieldname}::dump(level, msg);\n"
-                dump_func = dump_func.format(
-                    indent = self._indent_string(),
-                    fieldname = field_cxx_name,
-                )
-        outfile.write(dump_func)
-        self._decrease_indent()
+            else:
+                self._generate_sysreg_64_get(outfile, reg)
+                self._generate_sysreg_64_set(outfile, reg)
 
-        outfile.write(self._indent_string())
-        dump_func = "}\n"
-        outfile.write(dump_func)
+            self._generate_register_fieldsets(outfile, reg)
 
-        outfile.write("}\n")
-        self._decrease_indent()
+        _in_register_namespace(self, outfile, reg)
 
-    def _generate_fieldsets(self, reg, outfile):
-        if not reg.is_sysreg: return
+    def _generate_register_fieldsets(self, outfile, reg):
+        """
+        Generate accessors for all fieldsets that belong to a register
+        """
         fieldsets = reg.fieldsets
 
         if len(fieldsets) == 1:
-            self._generate_single_fieldset(reg, fieldsets[0], outfile)
-        elif len(reg.fieldsets) > 1:
+            self._generate_fieldset(outfile, reg, fieldsets[0])
+
+        elif len(fieldsets) > 1:
             for idx, fieldset in enumerate(fieldsets):
-                outfile.write("\n")
-
-                if(fieldset.condition is not None):
```

Comment:
We lost this fieldset condition comment block and I think that is a useful feature to have in the header. Maybe put this block back in or create a gadget that can append fieldsets with their condition description.

Reply:
Good catch, this got removed by accident. Will add back